### PR TITLE
Reset pause button when content video reaches end

### PR DESCRIPTION
**Pull request by swemail**
This is so that host can see that a meditation, reflection or sharing that is not looping has reached it's end and simply press play again

### CODE REVIEW

File: client/src/routes/Session/components/Slides/Slides/Blocks/Video.tsx [hunk 1]
```diff
@@ -39,17 +44,19 @@ const Video: React.FC<VideoProps> = ({
...
   const [duration, setDuration] = useState(0);
   const exerciseState = useSessionState(state => state.session?.exerciseState);
   const previousState = useRef({playing: false, timestamp: new Date()});
 
-  const seek = (seconds: number) => {
+  const seek = useCallback((seconds: number) => {
     videoRef.current?.seek(seconds);
     timerRef.current?.seek(seconds);
-  };
+  }, []);
```
Comment (gewfy) [inline client/src/routes/Session/components/Slides/Slides/Blocks/Video.tsx:59]: Nice catch 😅

**Review by gewfy — APPROVED** [review]
Tight code! Wouldn't this cause everyone to stop when the host stops? If the clients are a bit out of sync, maybe better to not change the exercise state and instead just visually change it? Could be a non issue ofc... so let's test!

**Comment by swemail** [comment]
> Tight code! Wouldn't this cause everyone to stop when the host stops? If the clients are a bit out of sync, maybe better to not change the exercise state and instead just visually change it? Could be a non issue ofc... so let's test!

Probably true, but wouldn't the time compensation code kind of make up for that? It's not exact of course. Just setting it visually to true would have us toggle pause on and off via the server to start playing again. Don't know what is wisest here

**Comment by gewfy** [comment]
> > Tight code! Wouldn't this cause everyone to stop when the host stops? If the clients are a bit out of sync, maybe better to not change the exercise state and instead just visually change it? Could be a non issue ofc... so let's test!
> 
> Probably true, but wouldn't the time compensation code kind of make up for that? It's not exact of course. Just setting it visually to true would have us toggle pause on and off via the server to start playing again. Don't know what is wisest here

I still think we should test your approach and see if it would cut other users off. But I don't trust the time dilation :P And I haven't finished this thought so I might be wrong... If we only change it visually for the host. And then when the host press play again it's just setting playing = true again (exercise state is already true) which is a reset.

**Comment by swemail** [comment]
> > > Tight code! Wouldn't this cause everyone to stop when the host stops? If the clients are a bit out of sync, maybe better to not change the exercise state and instead just visually change it? Could be a non issue ofc... so let's test!
> > 
> > 
> > Probably true, but wouldn't the time compensation code kind of make up for that? It's not exact of course. Just setting it visually to true would have us toggle pause on and off via the server to start playing again. Don't know what is wisest here
> 
> I still think we should test your approach and see if it would cut other users off. But I don't trust the time dilation :P And I haven't finished this thought so I might be wrong... If we only change it visually for the host. And then when the host press play again it's just setting playing = true again (exercise state is already true) which is a reset.

I'm close to a simpler local solution, will try to make that work. It's better anyways and as yous aid, just setting it to play again is a reset

File: client/src/routes/Session/components/Slides/Slides/Content.tsx [hunk 2]
```diff
@@ -58,8 +58,8 @@ const Content: React.FC<ContentProps> = ({slide: {content = {}}, active}) => {
   );
 
   const resetCallback = useCallback(() => {
-    setPlaying(false);
-  }, [setPlaying]);
+    setCurrentContentReachedEnd(true);
+  }, [setCurrentContentReachedEnd]);
```
[the remark this answers is not in the document]
Comment (swemail) [reply client/src/routes/Session/components/Slides/Slides/Content.tsx:62]: Yes, makes sense, will fix

**Pull request merged**